client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/configure-email-notification/index.tsx

```diff
@@ -1,39 +1,49 @@
 import { Card, Button } from '@automattic/components';
 import { CheckboxControl } from '@wordpress/components';
-import { Icon, plus } from '@wordpress/icons';
+import { Icon, plus, pencil } from '@wordpress/icons';
 import { useTranslate } from 'i18n-calypso';
 import { useEffect, useState } from 'react';
+import Badge from 'calypso/components/badge';
+import type { MonitorSettingsEmail } from '../../sites-overview/types';
 
 import './style.scss';
 
-interface EmailItem {
-	email: string;
-	name: string;
+interface StateEmailItem extends MonitorSettingsEmail {
 	checked: boolean;
+	isDefault?: boolean;
 }
 interface Props {
 	defaultEmailAddresses: Array< string >;
 	toggleModal: () => void;
+	addedEmailAddresses?: Array< MonitorSettingsEmail >;
 }
 
 export default function ConfigureEmailNotification( {
 	defaultEmailAddresses = [],
 	toggleModal,
+	addedEmailAddresses = [], // FIXME: This value will come from the API.
 }: Props ) {
 	const translate = useTranslate();
 
-	const [ allEmailItems, setAllEmailItems ] = useState< EmailItem[] >( [] );
+	const [ allEmailItems, setAllEmailItems ] = useState< StateEmailItem[] >( [] );
 
 	useEffect( () => {
 		if ( defaultEmailAddresses ) {
-			const allEmailItems = defaultEmailAddresses.map( ( email, index ) => ( {
+			const defaultEmailItems = defaultEmailAddresses.map( ( email ) => ( {
 				email,
-				checked: index === 0, // FIXME: This should be dynamic.
 				name: 'Default Email', //FIXME: This should be dynamic.
+				checked: true,
+				isDefault: true,
+				verified: true,
 			} ) );
-			setAllEmailItems( allEmailItems );
+			const addedEmailItems = addedEmailAddresses.map( ( email ) => ( {
+				...email,
+				checked: email.verified, // Checked only if the email is verified.
+			} ) );
+			setAllEmailItems( [ ...defaultEmailItems, ...addedEmailItems ] );
 		}
-	}, [ defaultEmailAddresses ] );
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
```

Comment:
Just curious why we don't add `defaultEmailAddresses` and `addedEmailAddresses` as dependencies here. Is it because this is a temporary logic?

Reply:
Yes, this is just a temporary logic.